lardataobj/RawData/raw.cxx

```diff
@@ -1173,4 +1200,207 @@ namespace raw {
+          else if (d==21) {add_to_sequence_terminate({0,0,0,0,0,0,1}, sqce, cmp); break;}
+          else if (d==22) {add_to_sequence_terminate({1,0,0,0,0,0,1}, sqce, cmp); break;}
+          else if (d==23) {add_to_sequence_terminate({0,1,0,0,0,0,1}, sqce, cmp); break;}
+          else if (d==24) {add_to_sequence_terminate({0,0,1,0,0,0,1}, sqce, cmp); break;}
+          else if (d==25) {add_to_sequence_terminate({1,0,1,0,0,0,1}, sqce, cmp); break;}
+          else if (d==26) {add_to_sequence_terminate({0,0,0,1,0,0,1}, sqce, cmp); break;}
+          else if (d==27) {add_to_sequence_terminate({1,0,0,1,0,0,1}, sqce, cmp); break;}
+          else if (d==28) {add_to_sequence_terminate({0,1,0,1,0,0,1}, sqce, cmp); break;}
+          else if (d==29) {add_to_sequence_terminate({0,0,0,0,1,0,1}, sqce, cmp); break;}
+          else if (d==30) {add_to_sequence_terminate({1,0,0,0,1,0,1}, sqce, cmp); break;}
+          else if (d==31) {add_to_sequence_terminate({0,1,0,0,1,0,1}, sqce, cmp); break;}
+        }
+      
+        // Find the first Fibonacci number that exceed the number
+        for (; iFib<FibNumbers.size(); ++iFib) {
+          if (d < FibNumbers.at(iFib)) {
+            break;
+          }
+        }
+
+        // And subtract the previous one from the initial number
+        d = d - FibNumbers.at(iFib-1);
+        // Store the previous Fibonacci number
+        if (!sqce.size()) {
+          sqce = std::vector<bool>(iFib-1,0);
+          sqce.at(iFib-2) = 1;
+        } else {
+          sqce.at(iFib-2) = 1;
+        }
+      }
+    }
+  
+    // Now convert all this to a vector of short and it's just another day in paradise for larsoft
+    size_t n_vector = cmp.size();
+    while (n_vector>sizeof(short)*8) {
+      // Create a bitset of the size of the short to simplify things
+      std::bitset<8*sizeof(short)> this_encoded;
+      // Set the bitset to match the stream
+      for (size_t it=0; it<8*sizeof(short); ++it) {
+        if (cmp[it]) this_encoded.set(it);
+      }
+      // Get rid of stuff we've dealt with 
+      cmp.erase(cmp.begin(), cmp.begin()+8*sizeof(short));
+      // Cast the bitset to short
+      short comp_s = (short)this_encoded.to_ulong();
+    
+      // Store the short in the output waveform
+      comp_short.push_back(comp_s);
+      n_vector = cmp.size();
+    }
+  
+    // Deal with the last part
+    std::bitset<8*sizeof(short)> this_encoded;
+    for (size_t it=0; it<cmp.size(); ++it) {
+      if(cmp[it]) this_encoded.set(it);
+    }
+    short comp_s = (short)this_encoded.to_ulong();
+    comp_short.push_back(comp_s);
+    wf = comp_short;
+    return;
+  }
+
+  void UncompressFibonacci(const std::vector<short> &adc,
+                           std::vector<short>       &uncompressed,
+                           std::map<int, short>      FibNumbers) {
+
+    if (!FibNumbers.size()) FibNumbers = GetFibonacciTable();
+    // First compressed sample is the size
+    size_t n_samples = adc[0];
```

Comment:
Here's the short n_samples from adc[0] -- maybe more?  The ZS compression method is similarly limited (though it is quite old and perhaps should be updated).  In the MC, we don't run long samples however, but we might want to re-compress data.